azure-pipelines.yml

```diff
@@ -787,26 +787,29 @@ jobs:
       name: 'linux-pool'
       demands: assignment -equals default
     variables:
-      pr.num: $[ variables['System.PullRequest.PullRequestNumber'] ]
       branch_sha: $[ dependencies.git_sha.outputs['out.branch'] ]
       status: $[ dependencies.collect_build_data.result ]
       is_release: $[ dependencies.check_for_release.outputs['out.is_release'] ]
     steps:
+      - checkout: self
+        persistCredentials: true
+      - template: ci/bash-lib.yml
+        parameters:
+          var_name: bash-lib
       - bash: |
           set -euo pipefail
 
-          tell_daml() {
-              local message pr_handler
-              message=$1
-              pr_handler=$(head -1 release/rotation | awk '{print $1}')
-              curl -XPOST \
-                   -i \
-                   -H 'Content-Type: application/json' \
-                   --data "{\"text\":\"<@${pr_handler}> <https://dev.azure.com/digitalasset/daml/_build/results?buildId=$(Build.BuildId)|Build $(Build.BuildId)> for release PR <https://github.com/digital-asset/daml/pull/$(pr.num)|#$(pr.num)> has completed with status ${message}.\"}" \
-                   $(Slack.team-daml)
-          }
+          source $(bash-lib)
+
+          AUTH="$(get-gh-auth-header)"
+          PR=$(curl -H "$AUTH" \
```

Comment:
Maybe add a `-f` in here so it fails properly.

Reply:
It doesn't, though, because Bash doesn't propagate errors through pipes. You just end up with an empty `PR`. Maybe I should check for that? 🤔

Reply:
It does if you `set -o pipefail`, which you have, doesn't it?

Reply:
Not in a subshell. `-o pipefail` works for pipes in the current script but in my experience (and I tested it for this one) it doesn't in subshells. Or maybe it is that Bash doesn't propagate errors through string interpolation? Not quite sure what the actual issue is, but it does succeed on that statement and sets the variable to an empty string. (I think; now that I say it I did not try to differentiate between empty string and unset.)

Reply:
```
bash-3.2$ set -euo pipefail
bash-3.2$ echo "hello $(exit 1) world"
hello  world
bash-3.2$ 
```

Reply:
🤔
The more I test this the more confused I get.

Reply:
So Bash does seem to fil as expected now. I'm pretty sure I tested it about an hour ago and it didn't, but 🤷 . Thinking more about it, though, in this case, I do not want it to fail, as the Slack notification is still useful even if the script does not manage to get the PR number.

Reply:
Subshells are not pipes.

```
$ ((exit 1) | echo 'hello, world'); echo $?
hello, world
0
$ (set -o pipefail; (exit 1) | echo 'hello, world'); echo $?
hello, world
1
```

Reply:
That said, I like your fix that just reports anyway. 🙂